aten/src/ATen/native/transformers/cuda/sdp_utils.h

```diff
@@ -162,6 +164,25 @@ inline bool check_head_dim_size(sdp_params params, bool debug) {
   return true;
 }
 
+inline bool check_head_dim_size_mem_efficient(sdp_params params, bool debug) {
```

Comment:
does this code not need to be traced, so it doesn't need to be symintified? (e.g. it's only invoked somewhere inside a cuda kernel)

Reply:
hmm thats a good question: 
This code is used to check if we can run the fused kernels:
https://github.com/pytorch/pytorch/blob/ed41a7fb681ff51129763213042da0be4fbd1a56/aten/src/ATen/native/transformers/cuda/attention.cu#LL779C29-L779C29

In particular it is uesd in the cuda implementation of the native function _fused_sdp_choice which is in turn used here:
https://github.com/pytorch/pytorch/blob/ed41a7fb681ff51129763213042da0be4fbd1a56/aten/src/ATen/native/transformers/attention.cpp#L693

Reply:
yep - you're right, it looks like this code doesn't need to be traced (it's only called from inside of cuda kernels)